quintilesims/layer0
# [WIP] 500: Set launch type for services and tasks

**What does this pull request do?**
Services and tasks now check the environment's type (static or dynamic) and use that value to determine the type of instance (Fargate or EC2) on which they should be deployed.


**How should this be tested?**
Spin up services and tasks in both dynamic and static environments, and confirm that dynamic environments yield Fargate instances and that static environments yield EC2 instances. Also, run unit tests at least for the `provider` package.

**Notes:**
- Fargate is only available in `us-east-1` right now, so make sure that's set in `l0-setup`.
- I've pushed an API binary to dockerhub: specify `tlaketest` in `l0-setup`.
- There are some requirements for a task definition file to be Fargate-compatible. Probably the most notable are the following four parameters (defined at the same level as `"containerDefinitions"`):
```
    "requiresCompatibilities": [ "FARGATE" ], 
    "networkMode": "awsvpc", 
    "cpu": "256", 
    "memory": "512",
```
More info at https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html - in particular, note that there are specific values allowed for `"cpu"` and `"memory"`.

**Checklist**


closes #500
